src/pal/src/pal_api.c

```diff
@@ -104,11 +104,24 @@ int occlum_pal_init(const struct occlum_pal_attr *attr) {
     eid = pal_get_enclave_id();
 
     int ecall_ret = 0;
-    const char *resolv_conf_ptr = pal_load_resolv_conf();
+    struct host_file_buffer file_buffer = {
+        .hostname_ptr = pal_load_file("/etc/hostname"),
+        .hosts_ptr = pal_load_file("/etc/hosts"),
+        .resolv_conf_ptr = pal_load_file("/etc/resolv.conf"),
+    };
+
+    const struct host_file_buffer *file_buffer_ptr = &file_buffer;
+
     sgx_status_t ecall_status = occlum_ecall_init(eid, &ecall_ret, attr->log_level,
-                                resolved_path, resolv_conf_ptr);
-    free((void *)resolv_conf_ptr);
-    resolv_conf_ptr = NULL;
+                                resolved_path, file_buffer_ptr);
+
```

Comment:
Why not implement a function to implement the free of the `host_file_buffer`?

Reply:
Done